### ISSUE TITLE
How Can One Fine-Tune or Continue to Train a Pre-Trained Model

### ISSUE THREAD

**Issue by II-Matto** [dmlc/mxnet#379]
If I have a pre-trained model and want to fine-tune it with new data, or continue to train it for more iterations, what should I do?

**Comment by tqchen** [comment]
You can feed the model parameters into a new FeedforwardModel https://github.com/dmlc/mxnet/blob/master/python/mxnet/model.py#L492 and call fit again.

**Comment by II-Matto** [comment]
Another two questions:

1. Loss for regression: I figured out that there are sigmoid output and linear ouput. Are they used for regression (Euclidean loss)? Can I set multi-dimensional real-valued labels for data points?

2. Multi-loss: I want to have several different losses for one network, e.g. k outputs for classification with softmax loss and m outputs for regression with Euclidean loss. How can I achieve this?

**Comment by tqchen** [comment]
- Yes, you can feed multi dimensional real value labels in.
- Multi-loss: We haven't roll this out in the Feedforward API, mainly due to the first wave change to bring the API ready for the easiest task, but only minor changes are needed to be make to model training code to achieve this (mainly for copying the labels to right position).
   - also see the rnn example, which contains multiple loss
   - Please open a issue on this, or even better consider contributing:)

**Comment by II-Matto** [comment]
Thanks for your quick response!

One more question. Is it possible to add weights for each output of each sample? So that different samples are used for different loss. One sample doesn't contribute to the loss of a output unit when the corresponding weight is zero.

**Comment by tqchen** [comment]
Yes, this is possible. It is not supported by current symbolic operators, however, we can create a new loss function that accepts the weight(1 length tensor as argument) and customize that behavior.

If the weight is very non-uniform(say, on scale of [0,100] ), it is not going to work, as SGD is not very good at deal with non uniformly scaled gradient, randomly drop the examples by probability weight might be a better idea

**Comment by II-Matto** [comment]
Yes, the weights shoud fall into a reasonable range or there might be optimization problems.

The weights can be useful when there are missing labels. One can still use all the samples and each sample contributes to a loss only when it has the corresponding label.

It can also help in case of unbalanced training set. When there are fewer samples for one class, one may want to assign larger weights to them.

In a word, the weights will be useful in maintaining balance both between samples and between losses.

**Comment by II-Matto** [comment]
BTW, how can it be achieved that model be saved every k iterations? Currently I save a check point after each iteration.

**Comment by tqchen** [comment]
modify this https://github.com/dmlc/mxnet/blob/master/python/mxnet/callback.py#L12 add a period parameter (with default=1)

**Comment by tqchen** [comment]
I will close this issue for now, please feel free to reopen it if there is future questions on this thread

**Comment by antonysamuelb** [comment]
Hi,
I am using MXNET in R. I am trying linear regression output. But I have no convergence when I give more than 2 hidden layers. So I decided to go for pre-training using autoencoders. 
I have initial weights that I obtained by training a denoising autoencoder (from autoencoder package). How do I initialize a model with those weights?
